### DISCUSSION TITLE
Tests tracing in laravel 11

### DISCUSSION THREAD

**Discussion by vaniok010** [laravel/framework#51154] [General]
In Laravel 10, if I make a mistake in a test, I get a trace. In Laravel 11, this is not the case. Why and how to return it?

I installed 2 new projects with laravel 10 and laravel 11

tracing in Laravel 10

```
Runtime: PHP 8.3.6 Configuration: /home/ivan/www/laravel10.loc/phpunit.xml

.F 2 / 2 (100%)

Time: 00:00.085, Memory: 28.00 MB

There was 1 failure:

Tests\Feature\ExampleTest::test_the_application_returns_a_successful_response Expected response status code [200] but received 500. Failed asserting that 500 is identical to 200.
The following exception occurred during the last request:

Error: Call to undefined function App\Http\Controllers\randomint() in /home/ivan/www/laravel10.loc/app/Http/Controllers/TestController.php:9 Stack trace: #0 /home/ivan/www/laravel10.loc/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): App\Http\Controllers\TestController->test() #1 /home/ivan/www/laravel10.loc/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(43): Illuminate\Routing\Controller->callAction() #2 /home/ivan/www/laravel10.loc/vendor/laravel/framework/src/Illuminate/Routing/Route.php(259): Illuminate\Routing\ControllerDispatcher->dispatch() #3 /home/ivan/www/laravel10.loc/vendor/laravel/framework/src/Illuminate/Routing/Route.php(205): Illuminate\Routing\Route->runController() #4 /home/ivan/www/laravel10.loc/vendor/laravel/framework/src/Illuminate/Routing/Router.php(806): Illuminate\Routing\Route->run() ...
```

tracing in Laravel 11

```
Runtime: PHP 8.3.6 Configuration: /home/ivan/www/laravel11.loc/phpunit.xml

.F 2 / 2 (100%)

Time: 00:00.094, Memory: 30.00 MB

There was 1 failure:

Tests\Feature\ExampleTest::test_the_application_returns_a_successful_response Expected response status code [200] but received 500. Failed asserting that 500 is identical to 200.
/home/ivan/www/laravel11.loc/vendor/laravel/framework/src/Illuminate/Testing/TestResponse.php:159 /home/ivan/www/laravel11.loc/tests/Feature/ExampleTest.php:17

FAILURES! Tests: 2, Assertions: 2, Failures: 1.
```

**Comment by ahinkle** [comment]
Please provide more of your code or a replicable example for us to help.

**Reply by vaniok010** [reply]
1. Installing laravel 11
2. Make a mistake in the code, example: call a function that does not exist. In routes/web.php

```
<?php

use Illuminate\Support\Facades\Route;

Route::get('/', function () {
    return functionNotExist();
});
```
3. Run test

```
<?php

namespace Tests\Feature;

// use Illuminate\Foundation\Testing\RefreshDatabase;
use Tests\TestCase;


class ExampleTest extends TestCase
{
    /**
     * A basic test example.
     */
    public function test_the_application_returns_a_successful_response(): void
    {
        $response = $this->get('/');

        $response->assertStatus(200);
    }
}
```
4. No trace in the command line

![image](https://github.com/laravel/framework/assets/6134702/663b14a0-5657-487f-94ca-161b0b195706)

It's all in Laravel 10, it shows the trace

![image](https://github.com/laravel/framework/assets/6134702/7441ccc4-b2f7-4a68-9959-edf72884bfb3)

**Reply by ahinkle** [reply]
Actually, I'm encountering this too now. I see what you are referring to. Did you find a solution?

**Reply by vaniok010** [reply]
In Laravel 10, the runTest method was overridden in the base TestCase, in Laravel 11, this method was removed. You can add this method, but only with phpunit version 10. In phpunit version 11, the runTest method is declared final

![image_2024-04-26_14-09-41](https://github.com/laravel/framework/assets/6134702/886fc22b-810d-485c-bce6-85e2b1f85587)

phpunit v10
![image_2024-04-26_14-15-19](https://github.com/laravel/framework/assets/6134702/121592bc-c087-488d-95ca-4343e24d9b67)

phpunit v11
![image_2024-04-26_14-13-58](https://github.com/laravel/framework/assets/6134702/4af42de4-13c4-4409-9c6a-8e6be0dddd9e)

**Reply by ahinkle** [reply]
Gotcha. Looks like it's associated with the change in #48733

**Reply by AlliBalliBaba** [reply]
I think the problem is that [PHPUnit](https://github.com/sebastianbergmann/phpunit/blob/main/src/Framework/TestCase.php) will only call `transformException()` if an actual unintended Exception happens and not if an assertion fails. So we actually never see the error from the response since only an assertion error is thrown.

It's probably necessary to explicitly add the exception to the message from `$response->assertStatus`, at least that's what I've been doing.

I'll look into a PR when I have time
